### PR TITLE
Fix: HEAD-241 resolve critical and high vulnerabilities

### PR DESCRIPTION
**Pull request by j-luong**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
This PR should resolve critical and high vulnerabilities.

#### Where should the reviewer start?
The most notable changes:
- Major version bump of `strip-ansi` dependency where we are now required to use CommonJS style imports.
  - Changes made in PR
- Major version bump of `snyk/dep-graph` dependency in `packages/snyk-fix` where Node 8 support is dropped.
  - CLI already meets these requrements
- Major version bump of `snyk-config` dependency which drops Node 8 support and targets es2018. 
  - CLI already meets these requirements

### CODE REVIEW

**Comment by j-luong** [comment]
> Warnings
> ⚠️	
> Since the CLI is unifying on a standard and improved tooling, we're starting to migrate old-style `import`s and `export`s to ES6 ones. A file you've modified is using either `module.exports` or `require()`. If you can, please update them to ES6 [import syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) and [export syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export). Files found:
> 
> * `src/cli/main.ts`
> * `src/lib/analytics/index.ts`
> * `src/lib/iac/test/v2/output.ts`
> * `test/jest/unit/lib/commands/fix/fix.spec.ts`
> * `test/jest/unit/lib/formatters/format-error-result-summary.spec.ts`
> * `test/jest/unit/lib/formatters/format-monitor-response.spec.ts`
> * `test/jest/unit/lib/formatters/format-test-meta.spec.ts`
> * `test/jest/unit/lib/formatters/legal-license-instructions.spec.ts`
> * `test/jest/unit/lib/formatters/remediation-based-format-issues.spec.ts`
> * `test/jest/unit/lib/formatters/show-fix-tip.spec.ts`
> * `test/jest/unit/lib/formatters/test/display-result.spec.ts`
> * `test/jest/unit/policy-display.spec.ts`
> * `test/jest/unit/snyk-code/snyk-code-test.spec.ts`
> * `test/tap/auth.test.ts`
> * `test/tap/cli.test.ts`
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [e1d9f5b](https://github.com/snyk/cli/commit/e1d9f5bd7e257b369f240674bc7d07fa52c59d66)

Per `strip-ansi@^6.0.0`, CommonJS imports are [required](https://github.com/chalk/strip-ansi/releases/tag/v6.0.0)

File: src/cli/main.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ import {
   NoSupportedSastFiles,
 } from '../lib/errors';
 import { IaCErrorCodes } from './commands/test/iac/local-execution/types';
-import stripAnsi from 'strip-ansi';
+import stripAnsi = require('strip-ansi');
```
Comment (darscan) [inline src/cli/main.ts:32]: FYI for anyone wondering why this change is here: https://github.com/chalk/strip-ansi/releases/tag/v6.0.0

**Review by darscan — APPROVED** [review]

**Review by andreeaneata — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -113,7 +113,7 @@
     "proxy-from-env": "^1.0.0",
     "rimraf": "^2.6.3",
     "semver": "^6.0.0",
-    "snyk-config": "4.0.0",
+    "snyk-config": "^5.0.0",
```
Comment (PeterSchafer) [inline package.json:116]: Suggestion: Personally I think that the usage of `^` is not a good practice as it means that we can't reproduce individual builds. Therefore I would like to propose that we use this chance to get rid of them step by step.

Comment (j-luong) [reply package.json:116]: Hmm, I'm not so sure, semantic versioning is pretty standard practice with npm packages and we include a `package-lock.json` that should make sure we have consistent builds. Is there a scenario where this isn't the case?

Comment (PeterSchafer) [reply package.json:116]: withdrawn :)

**Review by PeterSchafer — APPROVED** [review]

**Review by michelkaporin — APPROVED** [review]
